packages/headless/src/auth/saml-client.ts

```diff
@@ -0,0 +1,25 @@
+import {buildSamlProvider} from './saml-provider';
+
+export interface SamlClientOptions {
```

Comment:
I think we could drop the `Saml `keyword here, since we are already in a `saml-provider` module.

```
interface ClientOptions {}

interface Client {}
```

Reply:
I feel `Client` and `ClientOptions` are missing some reassurance hmm ...

If we had an npm package or sub-module called `@coveo/saml` or `@coveo/headless/saml`, then it might be ok,

```
import {buildClient, Client} from '@coveo/headless/saml`
```
but even then it feels too vanilla.

Maybe I need to time to adjust but I'm also thinking about how we expose `buildSearchEngine`, `buildRecommendationEngine`, from the different sub-packages, vs. just `buildEngine`.

On the other hand, if the package is a more generic `auth` package containing helpers for different approaches, then I think Saml is absolutely needed

```
import {buildSamlClient} from '@coveo/auth'
```